users/src/main/java/io/crate/operation/auth/AuthenticationService.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of a module with proprietary Enterprise Features.
+ *
+ * Licensed to Crate.io Inc. ("Crate.io") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ *
+ * Unauthorized copying of this file, via any medium is strictly prohibited.
+ *
+ * To use this file, Crate.io must have given you permission to enable and
+ * use such Enterprise Features and you must have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  If you enable or use the Enterprise
+ * Features, you represent and warrant that you have a valid Enterprise or
+ * Subscription Agreement with Crate.io.  Your use of the Enterprise Features
+ * if governed by the terms and conditions of your Enterprise or Subscription
+ * Agreement with Crate.io.
+ */
+
+package io.crate.operation.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.net.InetAddresses;
+import org.jboss.netty.handler.ipfilter.CIDR4;
+
+import javax.annotation.Nullable;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+public class AuthenticationService implements Authentication {
+
+    private static final String DEFAULT_AUTH_METHOD = "trust";
+    private static final String KEY_USER = "user";
+    private static final String KEY_ADDRESS = "address";
+    private static final String KEY_METHOD = "method";
+
+    private final Map<String, AuthenticationMethod> authMethodRegistry = new HashMap<>();
+    private Map<String, Map<String, String>> hbaConf;
```

Comment:
maybe the variable names could be changed a bit to make it clearer what the keys / values represent?